terraform/main.tf

```diff
@@ -0,0 +1,15 @@
+resource "juju_application" "mimir_worker" {
+  name = var.app_name
+  # Coordinator and worker must be in the same model
+  model = var.model_name
+  trust = var.trust
+
+  charm {
+    name    = "mimir-worker-k8s"
+    channel = var.channel
+  }
+  units  = var.units
+  config = var.config
+  # TODO: add revision, if given
```

Comment:
Looking at juju provider [docs](https://registry.terraform.io/providers/juju/juju/latest/docs/resources/application#nested-schema-for-charm), revision is to be added under `charm{}`. Maybe including the `TODO` there would be more clear

Reply:
Added :)